apollo-router/src/services/http/tests.rs

```diff
@@ -888,3 +1008,285 @@ async fn test_unix_socket() {
         .unwrap();
     insta::assert_json_snapshot!(response);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_alpn_negotiates_http2_when_both_support() {
+    let certificate_pem = include_str!("./testdata/server_self_signed.crt");
```

Comment:
I think the tests here that use the certs will start failing a year from now as the cert expires in 2026 according to `openssl x509 -in tls_localhost.crt -noout -dates`

In `apollo-router/src/services/http/tests.rs:235` there's another test that uses self signed certs with a note about expiry, but for that test its been set to 10 years. Can we reuse the certs from that test, and reference the comment about expiry here? Or if we can't use them for any reason, regenerating with a longer expiry date could be better